pubpub/pubpub
# dev: Apply css to pubs and add global className based on active element type

Okay, third time's the charm!

This does two simple things, resolving #1560:

- Adds custom scripts to Pubs
- Adds two classnames to the body of Pubs, Collections, and Pages (and no other objects):

1. `x-body-wrapper`. This is passed from the <Html> component in a new prop and is meant to describe the type of content on the page, primarily for allowing people to target their page layouts, generated app pages, and content (e.g. pubs) separately.

2. `active-[object]-[slug]`, where the object is `pub | collection | page` and the slug is the item's slug. This allows people to target specific Pubs, Collections, and Pages with CSS.

_Test Plan_
1. Make sure custom CSS is injected into Pubs
2. Check pages and make sure the body has the classes: 'layout-body-wrapper active-page-[slug]`
3. Check collections and make sure the body has the classes: 'layout-body-wrapper active-collection-[slug]`
4. Check pubs and make sure the body has the classes: 'pub-body-wrapper active-pub-[slug]`
5. Visit dashboard, login/logout, legal, and www pages and make sure they load and have no injected css or body class names.

_Post-Release_
Gabe to replace Commonplace and Contours CSS with a new version he made (with compiled scss!) that applies the styles only to pages so nothing changes.

I can see a future where we might want to add "app" body prefixes and inject css to community-owned/pubpub-controlled pages like Login/Logout/Signup/User/Legal. But for now, this feels like a good-enough extension.
